.github/workflows/build_publish_builder_image.yaml

```diff
@@ -0,0 +1,73 @@
+name: Build and Push Builder Image
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+    inputs:
+      go-version:
+        description: 'Go version'
+        required: false
+        default: ''
+      ubuntu-version:
+        description: 'Ubuntu version'
+        required: false
+        default: ''
+
+jobs:
+  build-and-push:
+    name: Build and Push Builder Image
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        go-version: [1.22.9, 1.23.2]
+        ubuntu-version: [16.04]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Set version
+        id: set_version
+        run: |
+          VERSION="${{ github.run_number }}-$(date +'%Y%m%d%H%M%S')"
```

Comment:
how the caller can select the very same data? should this be an input?

Reply:
It adds the workflow run number and date to the image tag. Renamed the variable to better reflect that.

Reply:
but I think the caller always uses latest, no?

Reply:
Simplified, only using latest now.